src/deploy/functions/build.ts

```diff
@@ -360,40 +348,43 @@ function discoverTrigger(endpoint: Endpoint): backend.Triggered {
     const bkEventFilters: Record<string, string> = {};
     for (const key in endpoint.eventTrigger.eventFilters) {
       if (typeof key === "string") {
-        bkEventFilters[key] = resolveString(endpoint.eventTrigger.eventFilters[key]);
+        bkEventFilters[key] = params.resolveString(
+          endpoint.eventTrigger.eventFilters[key],
+          paramValues
+        );
       }
     }
     const bkEvent: backend.EventTrigger = {
       eventType: endpoint.eventTrigger.eventType,
       eventFilters: bkEventFilters,
-      retry: resolveBoolean(endpoint.eventTrigger.retry),
+      retry: params.resolveBoolean(endpoint.eventTrigger.retry || false, paramValues),
     };
     if (endpoint.eventTrigger.serviceAccount) {
       bkEvent.serviceAccountEmail = endpoint.eventTrigger.serviceAccount;
     }
     if (endpoint.eventTrigger.region) {
-      bkEvent.region = resolveString(endpoint.eventTrigger.region);
+      bkEvent.region = params.resolveString(endpoint.eventTrigger.region, paramValues);
     }
     trigger = { eventTrigger: bkEvent };
   } else if ("scheduleTrigger" in endpoint) {
```

Comment:
Be very careful/explicit with the `in` operator. It's almost never what you want as an iterator (you probably want `of`) and it will return true here even if you said `endpoint.scheduleTrigger = undefined`. It'll also return true (IIRC) if the property was implemented in a base class.

Different codebases have different rules around membership tests.  The reigning standard is `{}.hasOwnProperty.call(endpoint, "scheduleTrigger")` but that's arcane as can be. We generally just check for truthiness unless we know the field is a number or bool.